### ISSUE TITLE
TypeVar when generic value must be of either type

### ISSUE THREAD

**Issue by siemib** [microsoft/pyright#331]
**Describe the bug**
When `TypeVar` is used to create a generic variable which is restricted to one of declared types pyright raises error. 

```
  11:21 - error: Expression of type 'Adder[Union[int, float]]' cannot be assigned to declared type 'Adder[int]'
  Type 'float' cannot be assigned to type 'int'
    'float' is incompatible with 'int'

```
**To Reproduce**
```
from typing import Generic, TypeVar

T = TypeVar('T', int, float)


class Adder(Generic[T]):
    def add(self, a: T, b: T) -> T:
        return a + b


adder: Adder[int] = Adder()
print(adder.add(1, 2))
```

**Expected behavior**
There should be no error issued.

**Comment by erictraut** [comment]
Thanks for the bug report. You're correct — PEP 484 indicates that this should work as you've indicated. I'll work on a fix.

**Comment by erictraut** [comment]
This will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.1.0, which I just published.
